core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -180,6 +180,10 @@ class SnappyStoreHiveCatalog(externalCatalog: SnappyExternalCatalog,
 
   /** API to get primary key or Key Columns of a SnappyData table */
   def getKeyColumns(table: String): Dataset[Column] = {
+    CatalogImpl.makeDataset(getKeyColumnsSeq(table), snappySession)
+  }
```

Comment:
Ok, checked the code and I was wrong about this. It is not used anywhere other than tests. Confused it with the listTables/listColumns functions in CatalogImpl that do return Datasets and perhaps whoever added this used the same template. I think it will be better to remove this and rename getKeyColumnsSeq to getKeyColumns as you suggested (and update the tests).

Reply:
@sumwale @vatsalmevada This gets used in CDC connector code(Although test but that is an example App too). If we are changing this then we need to change the code there as well and communicate others who may be using this API.
Please have a look at JavaCdcStreamingTestSuite.java

Reply:
Removed the API returning `Dataset` and renamed `getKeyColumnsSeq` to `getKeyColumns`. Also, this change leads to changes in snappy connectors repo. A separate PR is raised for the same.